src/main/java/com/metamx/metrics/cgroups/ProcCgroupDiscoverer.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2017 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metamx.metrics.cgroups;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Files;
+import com.metamx.common.RE;
+import com.metamx.common.StringUtils;
+import com.metamx.metrics.CgroupUtil;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ProcCgroupDiscoverer implements CgroupDiscoverer
+{
+  private static final String CGROUP_TYPE = "cgroup";
+  private static final String PROC_TYPE = "getProc";
+
+  @Override
+  public Path discover(final String cgroup, long pid)
+  {
+    Preconditions.checkNotNull(cgroup, "cgroup required");
+    // Wish List: find a way to cache these
```

Comment:
sure, I was torn with that because on one hand I don't like merging TODOs, but on the other I didn't have a great way to capture "hey there's probably an improvement here"